edb/pgsql/metaschema.py

```diff
@@ -1685,7 +1685,7 @@ class LocalDatetimeInFunction(dbops.Function):
                     )
                 )
             ELSE
-                val::timestamp
+                regexp_replace(val, '^\s*\+', '')::timestamp
```

Comment:
Use `to_timestamp(val, 'YYYY-MM-DDTHH:MI:SS.USZ')` instead.  That one is able to parse the sign and extended year digits.

Reply:
There's a problem with that one:
```
# SELECT to_timestamp('4368-04-02T04:26:41Z', 'YYYY-MM-DDTHH:MI:SS.USZ');
ERROR:  invalid value "Z" for "US"
DETAIL:  Value must be an integer.
```

Basically the ISO8601 standard makes the fractional seconds optional, while the specific parsing format makes them obligatory. That's why I didn't use it in the first place.

Reply:
Since we use the regexp anyway, we can detect the presence of optional parts and adjust the format string accordingly.

Reply:
Sigh, we can probably look at the seconds and if they have a `.` use one format and of they don't have fractional value, use another. This would come at the cost of one more "if" (I guess `CASE WHEN`).